### PR TITLE
Med 117 implement patient calendar UI

### PR DESCRIPTION
**Pull request by Mati-Pereira**
# Patient Card Calendar

**Calendar**
newcalendar

#### Features

 - Calendar UI for Patient Card
 
 **PR checklist:**

**1.1 Common**
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.1 types for input and output params
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url),
- [x] constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

 **1.3 Frontend|Mobile**
- [x] 1.3.1 split component and logic (move logic to custom hook file)
- [x] 1.3.2 colors, font size, and font name should be in the theme or in the constants
- [x] 1.3.3 no text in the components, use i18n approach, even if you have only one language for now
- [x] 1.3.4 inline styles prohibited
- [x] 1.3.5 import should be absolute. instead of ../../../components/myComponent should be components/myComponent
- [x] 1.3.6 unit tests for components

### CODE REVIEW

File: apps/medon-fe/src/components/PatientCardCalendar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      <Title>
+        <h2>{t('patient-card.calendar.title')}</h2>
+        {role === roles.local && (
+          <Link
+            to="#"
+            bgcolor={theme.colors.btnGradient}
+            textcolor={theme.colors.white}
+          >
+            {t('patient-card.calendar.link')}
+            <img src={Plus} alt="Plus svg" />
```
Comment (skazzp) [inline apps/medon-fe/src/components/PatientCardCalendar/index.tsx]: text to i18

Comment (Mati-Pereira) [reply apps/medon-fe/src/components/PatientCardCalendar/index.tsx]: fixed

**Review by skazzp — APPROVED** [review]

File: apps/medon-fe/src/components/PatientCardCalendar/hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { useState } from 'react';
+
+import { UseModalReturn } from './types';
+
+export function useModal(initialValue = false): UseModalReturn {
+  const [isVisible, setIsVisible] = useState(initialValue);
```
Comment (lyudad) [inline apps/medon-fe/src/components/PatientCardCalendar/hooks.tsx]: where types?

File: apps/medon-fe/src/components/PatientCardCalendar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  const [event, setEvent] = useState<Event>();
+
+  const theme = useTheme();
+  const { t } = useTranslation();
+  const { role } = useAppSelector((state) => state.userState.user);
+
+  const { hideModal, isVisible, showModal } = useModal(false);
+
+  const localizer = dayjsLocalizer(dayjs);
+  const dayPropGetter = getDayPropGetter(theme);
```
Comment (lyudad) [inline apps/medon-fe/src/components/PatientCardCalendar/index.tsx]: you will recalculate it with every rerender, find other way, same for line 40 and 42

**Comment by lyudad** [comment]
maybe I missed, but where are tests?

**Review by skazzp — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**